pkg/image/prune/prune.go

```diff
@@ -351,28 +366,33 @@ func addImageStreamsToGraph(g graph.Graph, streams *imageapi.ImageStreamList, li
 					}
 				}
 
+				if i == 0 {
+					glog.V(4).Infof("Adding edge (kind=%s) from %q to %q", kind, istNode.UniqueName(), imageNode.UniqueName())
+					p.g.AddEdge(istNode, imageNode, kind)
```

Comment:
Why do we need this?

Reply:
> Why do we need this?

In order to dereference imagestreamtag to image, there must be some kind of mapping between the two. It could be regular `map[istagname]image`. But since we already have the graph, we can utilize it. The fact is that `ImageStreamTagNode` takes unnecessarily too much memory - we could make the istnode a lot smaller by making it contain just a single string (istagname). Or would you prefer regular `map`?